### ISSUE TITLE
Old values in roles and permissions form after error

### ISSUE THREAD

**Issue by sergotail** [Laravel-Backpack/PermissionManager#230]
# Bug report

### What I did:
I have user editing page, where i can add or remove roles and permissions using built-in checkboxes. Also i have some user policy which forbids to add some roles or permissions to users. When such role or permission have been set during editing process, 403 page occurs.
### What I expected to happen:
When i go back, i expect to see all forms being filled with old values, and same submit button will send the same form data with errornous roles and permissions set.
### What happened:
When i go back, i see all forms being filled with old values, but the submit button sends no roles and permissions changes. This is confusing when some set checkboxes are ok, but only few of them set incorrectly. Intuitively i just uncheck all wrong checkboxes, but previously set right checkboxes are not sent.
### What I've already tried to fix it:
Nothing *lazy smile*
### Backpack, Laravel, PHP, DB version:
`backpack/crud                          4.0.59`
`backpack/generators                v2.0.7`
`backpack/permissionmanager 5.0.7`
`laravel/framework                    v7.5.2`
`php                                           7.3.9`

**Comment by pxpm** [comment]
@sergotail hello :) Thanks for submitting your issue. 

Please just give me some additional information so I can debug this properly.

1º - Are you using the default interface that comes with PermissionManager ?

2º - Instead of the 403, could you do a redirect back with input and a error message ?
On error: 
```php
\Alert::error('error message')->flash();
return redirect()->back()->withInput();
```

I think when you show the 403 page your request input will be lost.
You should redirect to the form with the errors so the `old()` helper will catch them and fill the form.

My guess is that this is **NOT** a bug, but a bad implementation.

I will wait for your clarifications before closing. 

Let me know if my assumptions are correct. 

Thanks

**Comment by sergotail** [comment]
@pxpm thank you for the reply. I checked your assumption and redirected back with input. I see this behaviour:

1. There are 3 roles: **User** (id=2), **Moderator** (id=3) and **Admin** (id=5).

2. Suppose i have an **Admin** role and decided to edit the roles of another user who has only the role **User**. I open the edit page for the selected user and see the checkbox only for the **User** role. That`s right.

3.  I added the **Admin** and **Moderator** roles to this user and removed the role **User**. Form-data sent:
```
roles[]: 3
roles[]: 5
roles_show[]: 3
roles_show[]: 5
```
3) My Policy declined this request because **Admin** can only add roles with lower power (i.e. **Moderator**)
4) I was redirected back to the edit page and now i see 3 checkboxes: for the roles **User**, **Moderator** and **Admin**. But i removed the role **User** on step 3. Ok, let`s send the request one more time not touching anything on the form. Form data sent:
```
roles[]: 5
roles[]: 3
roles_show[]: 2
roles_show[]: 3
roles_show[]: 5
```
It seems that there are no bugs in form, but a slightly confusing feature: on failure, both intended and already existed roles are shown, so it's hard to distinct them. Maybe it would be better to show previously set data entirely? I can't analyze use cases to determine the pros and cons of each solution. Anyway, there are no bug, sorry for the false alarm :)

**Comment by AbbyJanke** [comment]
@pxpm So I was looking into this to help out, so if you select permissions individually it works fine. But if you select a role say 'Admin' in the demo it keeps 'Admin' checked but unchecks all the permissions.

https://recordit.co/l0mA58Wrky

(sorry for the weird color scheme. Have a darkmode app I use based on my computer's dark mode settings.)

**Comment by pxpm** [comment]
@AbbyJanke and @sergotail 

I think we have fixed this in 4.1.28 with PR: https://github.com/Laravel-Backpack/CRUD/pull/3377

If you guys had a chance, and/or still need this can give it a try and let us know ? 

Wish you guys the best,
Pedro
